stock_available_to_promise_release/models/stock_move.py

```diff
@@ -75,21 +75,46 @@ def _promise_reservation_horizon_date(self):
         return None
 
     def _previous_promised_quantity_domain(self):
-        domain = [
-            ("need_release", "=", True),
+        """Lookup for product promised qty in the same warehouse.
+
+        Moves to consider are either already released or still be to released
+        but not done yet. Each of them should fit the reservation horizon.
+        """
+        base_domain = [
             ("product_id", "=", self.product_id.id),
-            ("date_priority", "<", self.date_priority),
             ("warehouse_id", "=", self.warehouse_id.id),
         ]
         horizon_date = self._promise_reservation_horizon_date()
         if horizon_date:
             # exclude moves planned beyond the horizon
-            domain.append(("date_expected", "<", horizon_date))
-        return domain
+            base_domain.append(("date_expected", "<=", horizon_date))
+
+        # either the move has to be released
+        # and priority is higher than the current one
+        domain_not_released = [
+            ("need_release", "=", True),
+            ("date_priority", "<", self.date_priority),
+        ]
+        # or it has been released already
+        # and the picking is printed
+        # and is not canceled or done
+        domain_released = [
+            ("need_release", "=", False),
+            ("picking_id.printed", "=", True),
```

Comment:
Why do we need to search what is printed? I don't understand the relation, why a stock.picking that we did not print would not be considered in the quantity and why a printed one would be considered?
Besides this, this would be a performance issue as the ORM would search all the picking_ids where printed is True first, which can be a lot of records after some time.

Reply:
this is what we came up to w/ Joël. The point is that now we have the split of the move + split of the picking, thus we should look for moves that are attached to pickings that are not processed yet. OTOH we can find a way to do it better.

Reply:
So on my side, I explain this with the way we account the moves to deduce from the current stock. So qty order to promise is:
 * Total in stock - Move to release (need_realease=True) - Move released not yet done (Printed=True as we set it on release now & state not done)

If @guewen you have any better domain for this I take it :) !